bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/discovery/EpsonProjectorDiscoveryService.java

```diff
@@ -0,0 +1,182 @@
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        ScheduledFuture<?> epsonDiscoveryJob = this.epsonDiscoveryJob;
+        if (epsonDiscoveryJob != null) {
+            logger.debug("Canceling background discovery job");
+            terminate = true;
+            epsonDiscoveryJob.cancel(false);
+            this.epsonDiscoveryJob = null;
+        }
+    }
+
+    @Override
+    public void startScan() {
+    }
+
+    @Override
+    public void stopScan() {
+    }
+
+    private synchronized void discover() {
+        logger.debug("Discovery job is running");
+
+        MulticastListener epsonMulticastListener;
+        NetworkAddressService networkAddressService = this.networkAddressService;
+
+        if (networkAddressService != null) {
+            String local = "127.0.0.1";
+            try {
+                String ip = networkAddressService.getPrimaryIpv4HostAddress();
+                epsonMulticastListener = new MulticastListener((ip != null ? ip : local));
+            } catch (SocketException se) {
+                logger.error("Discovery job got Socket exception creating multicast socket: {}", se.getMessage());
+                return;
+            } catch (IOException ioe) {
+                logger.error("Discovery job got IO exception creating multicast socket: {}", ioe.getMessage());
+                return;
+            }
+
+            while (!terminate) {
+                boolean beaconReceived;
+                try {
+                    // Wait for a discovery beacon.
+                    beaconReceived = epsonMulticastListener.waitForBeacon();
+                } catch (IOException ioe) {
+                    logger.debug("Discovery job got exception waiting for beacon: {}", ioe.getMessage());
+                    beaconReceived = false;
+                }
+
+                if (beaconReceived) {
+                    // We got a discovery beacon. Process it as a potential new thing
+                    Map<String, Object> properties = new HashMap<>();
+
+                    properties.put(THING_PROPERTY_HOST, epsonMulticastListener.getIPAddress());
+                    properties.put(THING_PROPERTY_PORT, DEFAULT_PORT);
+
+                    logger.trace("Projector with UID {} discovered at IP: {}", epsonMulticastListener.getUID(),
+                            epsonMulticastListener.getIPAddress());
+
+                    ThingUID uid = new ThingUID(THING_TYPE_PROJECTOR_TCP, epsonMulticastListener.getUID());
+                    logger.trace("Creating epson projector discovery result for: {}, IP={}", uid,
+                            epsonMulticastListener.getIPAddress());
+                    thingDiscovered(DiscoveryResultBuilder.create(uid).withProperties(properties)
+                            .withLabel("Epson Projector " + epsonMulticastListener.getUID()).build());
```

Comment:
Can you add the representation property?

Reply:
done